_docs/faq.markdown

```diff
@@ -0,0 +1,51 @@
+---
+title: FAQ
+order: 999999
+---
+
+## What's the progress on code completion?
+
+A decent amount of completion intelligence is
+already implemented, but there is definitely a lot more to do. Here are some
+examples of what works and what does not work (as of **June 20**).
+
+```rust
+use std::collections::HashMap;
+                        //^ completes to HashMap or HashSet
+
+struct S {
+    mapping: HashMap<i32, i32>
+            //^ knows that HashMap is in scope, so completes
+}
+
+fn main() {
+    let mut thing = S { mapping: HashMap::default() };
+                                          //^ no completion here yet :(
+
+    //v completes `thing`
+    thing.mapping.insert(92, 92);
+          //^ completes `mapping`
+                  //^ but does not know about `insert` yet :(
```

Comment:
Honestly, I don't know as I'm not hacking around completion stuff. I copied this from old docs.